dashboard/app/models/levels/blockly.rb

```diff
@@ -379,6 +380,33 @@ def localized_blockly_level_options(script)
     options.freeze
   end
 
+  def localized_blockly_level_options_for_lab2(script)
+    options = Rails.cache.fetch("#{cache_key}/#{script.try(:cache_key)}/#{I18n.locale}/localized_blockly_level_options_for_lab2", force: !Unit.should_cache?) do
+      level_options = blockly_level_options.dup
+
+      functions = level_options.
+        dig("levelData", "startSources", "blocks", "blocks")&.
+        filter {|block| block["type"] == "procedures_defnoreturn"}
+
+      functions&.each do |function|
+        function_name = function.dig("fields", "NAME")
+        next unless function_name
+
+        localized_name = I18n.t(
+          "name",
+          scope: [:data, :function_definitions, name, function_name],
+          default: function_name,
+          smart: true
+        )
+
+        function["fields"]["NAME"] = localized_name
+      end
+
+      level_options
```

Comment:
mino: should this just return `levelData` or even just `startSources` so the whole properties object doesn't have to be cached?

Reply:
Good question.  I know @mikeharv intends to do some more translation work so maybe we can see what it needs!